### PR TITLE
[cli] [workflow] Support "\r\n" -> "\n" convert in "ti format" command

### LINKED ISSUE #1675: [Format] Use Unix line endings ("\n")
**Describe the issue**

As we have more and more developers using different operating systems, line endings are becoming a somewhat annoying issue:
 - Windows uses `\r\n`
 - Unix (Linux and OS X) use `\n`

In fact, if you do `file **/*.cpp` under `/taichi/`, you'll find that we have been using different line endings for different files:
```
file **/*.cpp
analysis/alias_analysis.cpp:                 C source, ASCII text, with CRLF line terminators
analysis/build_cfg.cpp:                      C++ source, ASCII text, with CRLF line terminators
analysis/check_fields_registered.cpp:        C++ source, ASCII text
analysis/clone.cpp:                          C++ source, ASCII text, with CRLF line terminators
analysis/count_statements.cpp:               C++ source, ASCII text
analysis/data_source_analysis.cpp:           C source, ASCII text, with CRLF line terminators
analysis/detect_fors_with_break.cpp:         C++ source, ASCII text
analysis/gather_deactivations.cpp:           C++ source, ASCII text
analysis/gather_statements.cpp:              C++ source, ASCII text
...
```

("with CRLF line terminators" means `\r\n` is used, otherwise `\n`)

**Describe the solution you'd like (if any)**

I checked the LLVM and TensorFlow codebase, and they seem to be uniformly using Unix line endings (`\n`). **We should enforce Unix line endings as well.**

Here's what we should do
1. Update the `ti format` command to let it automatically convert all `\r\n` into `\n`.
2. Ensure the format checker fails on any appearance of `\r\n` in the codebase.

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #1675

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
It seems that our `format_plain_text` is already capable of "\n" convert. So I simply apply it to all file formats including .cpp and .py
Also removed the not-core-at-all `format` function from `core/util.py` B.T.W.
@xumingkuan Now, you may either request a review from @taichi-gardener or simply run `ti format` in CLI to fix CRLF issues.

### CODE REVIEW

File: python/taichi/code_format.py [hunk 1]
```diff
@@ -103,30 +108,31 @@ def find_diff_or_empty(s):
...
+                   style_config=os.path.join(repo_dir, 'misc', '.style.yapf'))
+    elif clang_format_bin and has_suffix(fn, ['cpp', 'h', 'cu', 'cuh']):
+        print('Formatting "{}"'.format(fn))
+        os.system('{} -i -style=file {}'.format(clang_format_bin, fn))
+    elif has_suffix(fn, ['txt', 'md', 'rst', 'cfg', 'll', 'ptx']):
+        print('Formatting "{}"'.format(fn))
+    else:
+        return False
+    format_plain_text(fn)
+    return True
```
Comment (yuanming-hu) [inline python/taichi/code_format.py:134]:
```suggestion
    if fn.endswith('.py'):
        print('Formatting "{}"'.format(fn))
        FormatFile(fn,
                   in_place=True,
                   style_config=os.path.join(repo_dir, 'misc', '.style.yapf'))
        return True
    elif clang_format_bin and has_suffix(fn, ['cpp', 'h', 'cu', 'cuh']):
        print('Formatting "{}"'.format(fn))
        os.system('{} -i -style=file {}'.format(clang_format_bin, fn))
        return True
    elif has_suffix(fn, ['txt', 'md', 'rst', 'cfg', 'll', 'ptx']):
        print('Formatting "{}"'.format(fn))
        format_plain_text(fn)
        return True
    else:
        return False
```

File: python/taichi/code_format.py [hunk 2]
```diff
@@ -103,30 +108,31 @@ def find_diff_or_empty(s):
...
-            pass
-        else:
-            print(f'Skipping {fn}...')
+        if not do_format(fn):
+            print(f'Skipping "{fn}"...')
 
     print('Formatting done!')
 
 
+def do_format(fn):
```
Comment (yuanming-hu) [inline python/taichi/code_format.py:119]: Rename to `format_file`?

File: python/taichi/code_format.py (hunk 1 above)
Comment (archibate) [reply python/taichi/code_format.py:134]: No, the \n formatter is (already) in format_plain_text, so we need to apply format_plain_text to .py and .cpp too.

Comment (yuanming-hu) [reply python/taichi/code_format.py:134]: I mean, please use `returns` in all branches for more clarity. (Updated the suggested change.)

**Comment by archibate** [comment]
@taichi-gardener functionality confirmed: https://github.com/taichi-dev/taichi/pull/1706/commits/dce7e3333ceca15b4075624aef1d66b7db5297ad

**Pull request merged**